### ISSUE TITLE
merged_model.pt is a sparse ZeRO-3 shard but 309/851 tensors are empty

### ISSUE THREAD

**Issue by penguinwang96825** [sediment1024/PhysRAG#3]
The released checkpoint at [sediment1024/PhysRAG](https://huggingface.co/sediment1024/PhysRAG) appears to be a rank-0 DeepSpeed ZeRO-3 state dict that was saved without gathering the sharded parameters. 309 of its 851 tensors are empty (i.e., `shape (0,)`).

This is not a download problem, because the SHA256 matches the published SHA256SUMS, and `physical_config.json` in the same repo already labels it "checkpoint_format": "zero3_rank0_sparse_state_dict"

```python
import torch
sd = torch.load("merged_model.pt", map_location="cpu")
empty = [k for k, v in sd.items() if v.numel() == 0]
print(len(sd), len(empty))          # 851 309
print(sd["module.blocks.0.injection.seq_proj.weight"].shape)   # torch.Size([0])
```

The gaps include PhysRAG's own injection parameters, in every injection layer:

```txt
module.blocks.0.injection.adapter_proj.weight  (3072, 16)   present
module.blocks.0.injection.adapter_proj.bias    (3072,)      present
module.blocks.0.injection.alpha                (1,)         present
module.blocks.0.injection.seq_proj.weight      (0,)         EMPTY
module.blocks.0.injection.seq_proj.bias        (4290,)      present
```

The empty tensors are mostly backbone weights (`blocks.N.ffn.*`, `blocks.N.self_attn.*`, `blocks.N.cross_attn.*`, 27 layers each), plus 30 of the 81 `blocks.N.original_block.*` entries.

Loading code that filters on `numel()`, which is necessary, otherwise `load_state_dict` fails on the shape mismatch, will load cleanly and run. The affected parameters simply keep whatever the freshly constructed module was initialised with. For `injection.seq_proj.weight` that is random initialisation, in all three injection layers. Inference then produces plausible videos and a plausible PhyGenBench score from a model that is only partly the trained one, with nothing in the output indicating it.

Is this the intended artifact, or was the checkpoint meant to be consolidated before release (e.g. via `zero_to_fp32.py` / `stage3_gather_16bit_weights_on_model_save`)?

**Comment by sediment1024** [comment]
Thanks for pointing this out and for the detailed investigation.
You are correct — the empty tensors were not intended. This appears to have been an oversight when exporting the checkpoint from our original DeepSpeed ZeRO-3 training run.
Unfortunately, the original training server and ZeRO-3 shards are no longer available, so we cannot reconstruct the fully consolidated version of this exact checkpoint. This is an issue with the released checkpoint artifact rather than the PhysRAG method or training procedure itself. For strict reproduction, we recommend training from the released code and configuration instead of treating the current merged_model.pt as a complete trained checkpoint.

**Comment by penguinwang96825** [comment]
@sediment1024 Thanks for the confirmation! We'll also consider and evaluate whether to run a training session.
